src/topology/locally_constant/algebra.lean

```diff
@@ -110,4 +110,27 @@ instance [ring Y] : ring (locally_constant X Y) :=
 instance [comm_ring Y] : comm_ring (locally_constant X Y) :=
 { .. locally_constant.comm_semiring, .. locally_constant.ring }
 
+variables (M : Type*) [monoid M] (A : Type*) [semiring A]
+
+instance : has_scalar M (locally_constant X M) :=
+{ smul := λ a f,
+  { to_fun := λ x, a*f(x),
+    is_locally_constant := is_locally_constant.comp (is_locally_constant f)
+      (has_mul.mul a) } }
```

Comment:
I think it would be nice to reuse `pi.has_scalar`:
```suggestion
  { to_fun              := m • (f : X → M),
    is_locally_constant :=
      by convert is_locally_constant.comp (is_locally_constant f) (has_mul.mul m), } }
```

Reply:
This generalizes to
```lean
instance [has_scalar R M] : has_scalar R (locally_constant X M) :=
{ smul := λ a f,
  { to_fun := a • f,
    is_locally_constant := ((is_locally_constant f).comp (has_scalar.smul a) : _) } }
```
after removing `monoid M` from the variables above.

Reply:
Oh nice, I was just about to see if I could excise the `convert`! Also agreed on the generalisation.